src/module.js

```diff
@@ -734,7 +734,11 @@ export default class DailyIframe extends EventEmitter {
 
     // if we're in callObject mode and not joined yet, don't do anything
     if (this._callObjectMode && this._meetingState !== DAILY_STATE_JOINED) {
-      return this;
+      return {
+        camera: { deviceId: this._preloadCache.videoDeviceId },
+        mic: { deviceId: this._preloadCache.audioDeviceId },
+        speaker: { deviceId: this._preloadCache.outputDeviceId },
+      };
     }
 
```

Comment:
Should we mark `setInputDevices()` as `async`, so that this gets converted to a Promise?

Reply:
whoops! good catch